packages/solid/src/reactive/signal.ts

```diff
@@ -1681,11 +1681,16 @@ function castError(err: unknown): Error {
   return new Error(typeof err === "string" ? err : "Unknown error", { cause: err });
 }
 
-function handleError(err: unknown): void {
+function runErrors(fns: ((err: any) => void)[], err: any) {
+  for (const f of fns) f(err)
+}
+function handleError(err: unknown) {
   const error = castError(err);
   const fns = ERROR && lookup(Owner, ERROR);
   if (!fns) throw error;
-  for (const f of fns) f(error);
+  if (Effects)
+    Effects!.push({ fn() { runErrors(fns, error); }, state: STALE } as unknown as Computation<any>);
```

Comment:
This is interesting. Im curios to learn more why we need to run them in effect 🤔